### DISCUSSION TITLE
How to turn a Parser<I, Parser<I, O>> into a Parser<I, O>?

### DISCUSSION THREAD

**Discussion by davnavr** [zesterer/chumsky#69] [Q&A] [answered]
I'm creating a parser for a toy language I'm working on, and I have a `HashMap<&'static str, BoxedParser<_, _, _>>` that I want to use to select a parser based on some prior keyword that was already parsed. Other libraries such as combine offer a [`then`](https://docs.rs/combine/4.6.2/combine/parser/trait.Parser.html#method.then) function or in FParsec for the F# language, the [`>>=`](http://www.quanttec.com/fparsec/reference/primitives.html#members.:62::62::61:) operator to select a parser based on input. Aside from trying to figure out how to use [`custom`](https://docs.rs/chumsky/latest/chumsky/primitive/fn.custom.html), is what I'm trying to do currently possible?

Thanks!

**Comment by zesterer** [comment, marked as answer]
Yes, this is now permitted. The `master` branch contains the [`Parser::then_with`](https://github.com/zesterer/chumsky/blob/master/src/lib.rs#L642-L665) combinator that should do what you want. It is due for inclusion in the next stable release.
